code/modules/projectiles/guns/projectile/constructable/railgun.dm

```diff
@@ -16,206 +16,173 @@
 			overlays += capacitor
 		else
 			var/image/capacitor = image('icons/obj/weaponsmithing.dmi', src, "railgun_capacitor_overlay")
 			overlays += capacitor
 
-/obj/item/weapon/gun/projectile/railgun/proc/update_verbs()
-	if(rod_loaded)
-		verbs += /obj/item/weapon/gun/projectile/railgun/verb/remove_rod
-	else
-		verbs -= /obj/item/weapon/gun/projectile/railgun/verb/remove_rod
-
-	if(capacitor)
-		verbs += /obj/item/weapon/gun/projectile/railgun/verb/remove_capacitor
-	else
-		verbs -= /obj/item/weapon/gun/projectile/railgun/verb/remove_capacitor
-
-	if(rails && !rails_secure)
-		verbs += /obj/item/weapon/gun/projectile/railgun/verb/remove_rails
-	else
-		verbs -= /obj/item/weapon/gun/projectile/railgun/verb/remove_rails
-
-/obj/item/weapon/gun/projectile/railgun/verb/remove_rod()
-	set name = "Unload railgun"
-	set category = "Object"
-	set src in range(0)
-
-	if(usr.isUnconscious())
-		to_chat(usr, "You can't do that while unconscious.")
-		return
-
-	if(!rod_loaded)
+/obj/item/weapon/gun/projectile/railgun/proc/remove_ammunition()
+	if(!loadedammo)
 		return
-	var/obj/item/I
-	if(rod_loaded == "Holy")
-		I = new /obj/item/weapon/nullrod(null)
-	else
-		I = new /obj/item/stack/rods(null)
-	I.forceMove(usr.loc)
-	usr.put_in_hands(I)
-	rod_loaded = 0
-	to_chat(usr, "You remove \the [I] from the barrel of \the [src].")
+	loadedammo.forceMove(usr.loc)
+	usr.put_in_hands(loadedammo)
+	to_chat(usr, "You remove \the [loadedammo] from the barrel of \the [src].")
+	loadedammo = null
 
 	update_icon()
-	update_verbs()
-
-/obj/item/weapon/gun/projectile/railgun/verb/remove_capacitor()
-	set name = "Unload capacitor bank"
-	set category = "Object"
-	set src in range(0)
-
-	if(usr.isUnconscious())
-		to_chat(usr, "You can't do that while unconscious.")
-		return
 
-	if(!capacitor)
+/obj/item/weapon/gun/projectile/railgun/proc/remove_capacitor()
+	if(!loadedcapacitor)
 		return
 
-	var/obj/item/weapon/stock_parts/capacitor/C = capacitor
-	C.forceMove(usr.loc)
-	usr.put_in_hands(C)
-	capacitor = null
-	to_chat(usr, "You remove \the [C] from the capacitor bank of \the [src].")
+	loadedcapacitor.forceMove(usr.loc)
+	usr.put_in_hands(loadedcapacitor)
+	to_chat(usr, "You remove \the [loadedcapacitor] from the capacitor bank of \the [src].")
+	loadedcapacitor = null
 
 	update_icon()
-	update_verbs()
-
-/obj/item/weapon/gun/projectile/railgun/verb/remove_rails()
-	set name = "Remove rail assembly"
-	set category = "Object"
-	set src in range(0)
-
-	if(usr.isUnconscious())
-		to_chat(usr, "You can't do that while unconscious.")
-		return
 
-	if(!rails)
+/obj/item/weapon/gun/projectile/railgun/proc/remove_rails()
+	if(!loadedassembly)
 		return
 
-	var/obj/item/weapon/rail_assembly/R = rails
-	R.forceMove(usr.loc)
-	usr.put_in_hands(R)
-	rails = null
-	to_chat(usr, "You remove \the [R] from the barrel of \the [src].")
-
+	loadedassembly.forceMove(usr.loc)
+	usr.put_in_hands(loadedassembly)
```

Comment:
no reason to rely on usr here instead of just passing a user, is there? you should also probably note you killed the verb in the CL

Reply:
knew i was forgetting something
i'll look into the usr thing